### PR TITLE
[jk] Allow canceling of individual running pipeline runs

### PR DESCRIPTION
**Pull request by johnson-mage**
# Description
- There was a button for canceling ALL running pipeline runs for an entire pipeline, which includes runs on different pages in the UI. This could confuse users if they select specific runs and think only the selected runs will be canceled (not all of them).
- This PR adds an "Actions" menu that consolidates the actions for retrying selected runs, canceling selected/checked pipeline runs in progress (new feature), and canceling all pipeline runs in progress.
- Clean up Pipeline Runs Table (rearrange columns, change date font size, change link colors, update logs icon).
- Fix positioning of "Select kernel" dropdown menu since it was cut off a bit.

# How Has This Been Tested?
retry and cancel selected runs
image

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [ ] Merge https://github.com/mage-ai/mage-ai/pull/3245 after it is ready, which will add completed block count to the `Block runs` column.

### CODE REVIEW

File: mage_ai/api/resources/PipelineResource.py [hunk 1]
```diff
@@ -312,16 +312,27 @@ def update_schedule_status(status, pipeline_uuid):
...
-                query.
-                filter(PipelineRun.pipeline_uuid == pipeline_uuid).
-                filter(PipelineRun.status.in_([
-                    PipelineRun.PipelineRunStatus.INITIAL,
-                    PipelineRun.PipelineRunStatus.RUNNING,
-                ]))
-            )
+        def cancel_pipeline_runs(
+            pipeline_uuid=None,
+            pipeline_runs=[],
```
Comment (dy46) [inline mage_ai/api/resources/PipelineResource.py:317]: let's add type hints

Comment (dy46) [reply mage_ai/api/resources/PipelineResource.py:317]: also don't default `pipeline_runs` to a empty list, you should set it to `None`

File: mage_ai/api/resources/PipelineResource.py [hunk 2]
```diff
@@ -340,7 +351,11 @@ def _update_callback(resource):
...
                     update_schedule_status(status, pipeline_uuid)
                 elif status == PipelineRun.PipelineRunStatus.CANCELLED.value:
-                    cancel_pipeline_runs(status, pipeline_uuid)
+                    pipeline_runs = payload.get('pipeline_runs')
+                    if pipeline_runs is None:
+                        cancel_pipeline_runs(pipeline_uuid=pipeline_uuid)
+                    else:
+                        cancel_pipeline_runs(pipeline_runs=pipeline_runs)
                 elif status == 'retry' and payload.get('pipeline_runs'):
                     retry_pipeline_runs(payload.get('pipeline_runs'))
```
Comment (dy46) [inline mage_ai/api/resources/PipelineResource.py:360]: nit: define `payload.get('pipeline_runs')` and reuse that

**Comment by johnson-mage** [comment]
@dy46 Do the rest of the BE changes seem fine? It was just the `mage_ai/api/resources/PipelineResource.py` file on the BE that was updated.

File: mage_ai/api/resources/PipelineResource.py [hunk 3]
```diff
@@ -312,16 +313,27 @@ def update_schedule_status(status, pipeline_uuid):
...
+            pipeline_uuid: str = None,
+            pipeline_runs: List[Dict] = None,
+        ):
+            if pipeline_runs:
+                pipeline_run_ids = [run.get('id') for run in pipeline_runs]
+                pipeline_runs = (
+                    PipelineRun.
+                    query.
+                    filter(PipelineRun.id.in_(pipeline_run_ids))
+                )
```
Comment (dy46) [inline mage_ai/api/resources/PipelineResource.py:326]: i would use a different variable for the result of the `PipelineRun.query`

**Pull request merged**